### PR TITLE
Basic transaction pool Implementation.

### PR DESCRIPTION
**Pull request by vldm**
Introduce basic transaction pool implementation.
Without real work with transaction.
Without real facilitator choosing algorithm.
Without real mixing.

But with real handling network event.

### CODE REVIEW

File: blockchain/src/block.rs [hunk 1]
```diff
@@ -97,10 +97,11 @@ pub struct KeyBlockHeader {
     /// Leader public key.
     pub leader: SecurePublicKey,
 
+    /// Facilitator of Transaction Pool.
+    pub facilitator: SecurePublicKey,
```
Comment (rtsisyk) [inline blockchain/src/block.rs]: Please include `facilitator` into Hashable for KeyBlockHeader.

File: src/console.rs [hunk 2]
```diff
@@ -109,21 +117,26 @@ where
         keys: &KeyChain,
         broker: Network,
         node: Node<Network>,
+        txpool: TransactionPoolManager,
```
Comment (rtsisyk) [inline src/console.rs]: Are you sure that you need TxPool intergation with Console?

File: txpool/protos/generated.proto [hunk 3]
```diff
@@ -0,0 +1,20 @@
+syntax = "proto3";
+package protobuf.pb;
+
+message SecurePublicKey {
```
Comment (rtsisyk) [inline txpool/protos/generated.proto]: I think that this approach with re-implementing protobuf encoding is flaky. You will need encoding of Transaction object. In order to implement it, about 80% of  node/protos should be copy-pasted. 

Let's extract encoding of points, UTXOs, blocks and transactions from `stegos_node` into `stegos_blockchain` or a separate crate. We will need protobuf for persistence too. Probably `stegos_blockchain` is appropriate place, but I'm not sure.

File: txpool/src/api.rs [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    ChangeFacilitator(secure::PublicKey),
+}
+
+//TODO: Pool should return some commulative transaction.
+#[derive(Debug, Eq, PartialEq)]
+pub enum PoolFeedback {
+    Ping,
+}
+
+pub struct TransactionPoolManager {
```
Comment (rtsisyk) [inline txpool/src/api.rs]: Typical convention here was `TransactionPool` for handler (API) and `TransactionPoolService` for Future.

File: txpool/src/protos/mod.rs [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+
+pub mod generated;
+pub mod messages;
+
+use stegos_crypto::pbc::secure;
+
+use failure::Error;
+use protobuf::Message as ProtobufMessage;
+
+pub trait StructIntoProto {
```
Comment (rtsisyk) [inline txpool/src/protos/mod.rs]: Feel free to apply this refactoring to node/protos/mod.rs (by a separate patch).

File: src/console.rs (hunk 2 above)
Comment (vldm) [reply src/console.rs]: We need to somehow test it for now

**Pull request merged**